transport/src/bin/bootnode.rs

```diff
@@ -107,7 +107,7 @@ async fn main() -> anyhow::Result<()> {
         .with_dns()?
         .with_behaviour(behaviour)
         .expect("infallible")
-        .with_swarm_config(|c| c.with_idle_connection_timeout(Duration::from_secs(120)))
+        .with_swarm_config(|c| c.with_idle_connection_timeout(Duration::from_secs(120))) // Suprisingly long but in itself not harmful esp. with the use of QUIC which reuses a single socket.
```

Comment:
What value would you suggest? And what factors should be considered when choosing it?

Reply:
Connections are kept-alive for as long as they are used. That is determine by at least 1 stream being open (i.e. a `Stream` instance exists, it is implemented using reference-counting).

Once all streams are closed, the timeout that you configure here kicks in.

This can be necessary if your application first dials a peer but doesn't use the connection immediately. Essentially, this value here should be long enough to cover the time period between dial and use of a connection. How long that is depends on the client that opens the connection. For example, if you have a client that first calls an API to dial a peer and then upon being notified that we are connected, computes something for 10s and then wants to send that data, this timeout should be > 10s.

As is said in the comment, with QUIC it really isn't a problem at all because all QUIC connections are multiplexed over a single socket so you are not going to hit an open FD limit. With TCP, it is more important to close idle connections.

If you have an upper connection limit, you should also take that into account when deciding how aggressively you want to close idle connections.

My guess would be, a value of 5-10 seconds should be more than enough here. That coupled with APIs for downstream clients that have a "dial if not connected" behaviour.

Reply:
We have computations that are taking up to 60s, but they're done as request-response. So if I understand correctly, connection wouldn't be considered idle if there's a response channel open, right?

Reply:
Correct, that is considered an active connection.

Reply:
If they aren't kept alive then that is a bug :)